src/query.ts

```diff
@@ -6,6 +8,7 @@ type Query = {
 type QueryOptions = {
   lookBackIndex: number
   multiWord: boolean
+  match: Match | null
```

Comment:
This is great! Much cleaner now. One small suggestion, we should probably call it `previousMatch` (or something similar), otherwise we try to pass `match` to get `match`, which is a bit confusing.

```suggestion
  previousMatch: Match | null
```

Reply:
done.